provider/resource_vault.go

```diff
@@ -125,22 +154,30 @@ func resourceVaultRead(data *schema.ResourceData, meta interface{}) error {
 
 	yamlString, err := cmd.CombinedOutput()
 	if err != nil {
-		log.Fatalf("ERROR [ansible-vault]: couldn't access ansible vault file %s with "+
-			"password file %s! %v", vaultFile, vaultPasswordFile, err)
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary: fmt.Sprintf("ERROR [ansible-vault]: couldn't access ansible vault file %s with "+
+				"password file %s! %v", vaultFile, vaultPasswordFile, err),
```

Comment:
Similar to ansible_playbook, we should use `yamlString` here. I would just replace this whole thing with something like:
```go
Summary: string(yamlString),
```
The error output I see when I do this looks like:
```
ansible_vault.ex: Refreshing state... [id=myvars.yml]
╷
│ Error: ERROR! Unable to read source file (myvars.yml): [Errno 2] No such file or directory: 'myvars.yml'
│ 
│ 
│   with ansible_vault.ex,
│   on main.tf line 9, in resource "ansible_vault" "ex":
│    9: resource "ansible_vault" "ex" {
│ 
│ ansible-playbook
╵
```
This looks pretty good to me.

Reply:
fixed